src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -1097,8 +1099,11 @@ private static string GetConfigurationAndPlatformPropertiesString(ProjectConfigu
         /// </summary>
         private static string GetPropertiesAttributeForDirectMSBuildTask(ProjectConfigurationInSolution projectConfiguration)
         {
-            string directProjectProperties = OpportunisticIntern.InternStringIfPossible(String.Join(";", GetConfigurationAndPlatformPropertiesString(projectConfiguration), SolutionProperties));
-            return directProjectProperties;
+            using SpanBasedStringBuilder directProjectProperties = Strings.GetSpanBasedStringBuilder();
+            directProjectProperties.Append(GetConfigurationAndPlatformPropertiesString(projectConfiguration));
+            directProjectProperties.Append(";");
```

Comment:
You're right, this is overkill. The advantage of doing it this way is that it's zero-alloc in case the resulting string has already been interned. But it costs significantly more time otherwise and is less readable so I've reverted it, thanks!